### PR TITLE
[feaLib] Support BASE table

### PR DESCRIPTION
**Pull request by khaledhosny**
Just the parts documented a currently implemented by Adobe’s
implementation.

### CODE REVIEW

File: Lib/fontTools/feaLib/testdata/spec9a.ttx [hunk 1]
```diff
@@ -0,0 +1,114 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<ttFont>
+
+  <BASE>
+    <Version value="1.0"/>
+    <HorizAxis>
+      <BaseTagList>
+        <!-- BaseTagCount=None -->
```
Comment (brawer) [inline Lib/fontTools/feaLib/testdata/spec9a.ttx]: makeotf produces `BaseTagCount=2` here; likewise for the other BaseTagCounts below.

File: Lib/fontTools/feaLib/testdata/spec9a.ttx [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+        <BaselineTag index="0" value="ideo"/>
+        <BaselineTag index="1" value="romn"/>
+      </BaseTagList>
+      <BaseScriptList>
+        <!-- BaseScriptCount=None -->
```
Comment (brawer) [inline Lib/fontTools/feaLib/testdata/spec9a.ttx]: makeotf produces `BaseScriptCount=6` here.

**Comment by brawer** [comment]
Very cool, thank you!

Here’s what makeotf produces for the test file in your unit test. The delta is only in comments, but since it’s probably only two lines to add, mind fixing them?

```xml
  <BASE>
    <Version value="1.0"/>
    <HorizAxis>
      <BaseTagList>
        <!-- BaseTagCount=2 -->
        <BaselineTag index="0" value="ideo"/>
        <BaselineTag index="1" value="romn"/>
      </BaseTagList>
      <BaseScriptList>
        <!-- BaseScriptCount=6 -->
        <BaseScriptRecord index="0">
          <BaseScriptTag value="cyrl"/>
          <BaseScript>
            <BaseValues>
              <DefaultIndex value="1"/>
              <!-- BaseCoordCount=2 -->
              <BaseCoord index="0" Format="1">
                <Coordinate value="-120"/>
              </BaseCoord>
              <BaseCoord index="1" Format="1">
                <Coordinate value="0"/>
              </BaseCoord>
            </BaseValues>
            <!-- BaseLangSysCount=0 -->
          </BaseScript>
        </BaseScriptRecord>
        <BaseScriptRecord index="1">
          <BaseScriptTag value="grek"/>
          <BaseScript>
            <BaseValues>
              <DefaultIndex value="1"/>
              <!-- BaseCoordCount=2 -->
              <BaseCoord index="0" Format="1">
                <Coordinate value="-120"/>
              </BaseCoord>
              <BaseCoord index="1" Format="1">
                <Coordinate value="0"/>
              </BaseCoord>
            </BaseValues>
            <!-- BaseLangSysCount=0 -->
          </BaseScript>
        </BaseScriptRecord>
        <BaseScriptRecord index="2">
          <BaseScriptTag value="hang"/>
          <BaseScript>
            <BaseValues>
              <DefaultIndex value="0"/>
              <!-- BaseCoordCount=2 -->
              <BaseCoord index="0" Format="1">
                <Coordinate value="-120"/>
              </BaseCoord>
              <BaseCoord index="1" Format="1">
                <Coordinate value="0"/>
              </BaseCoord>
            </BaseValues>
            <!-- BaseLangSysCount=0 -->
          </BaseScript>
        </BaseScriptRecord>
        <BaseScriptRecord index="3">
          <BaseScriptTag value="hani"/>
          <BaseScript>
            <BaseValues>
              <DefaultIndex value="0"/>
              <!-- BaseCoordCount=2 -->
              <BaseCoord index="0" Format="1">
                <Coordinate value="-120"/>
              </BaseCoord>
              <BaseCoord index="1" Format="1">
                <Coordinate value="0"/>
              </BaseCoord>
            </BaseValues>
            <!-- BaseLangSysCount=0 -->
          </BaseScript>
        </BaseScriptRecord>
        <BaseScriptRecord index="4">
          <BaseScriptTag value="kana"/>
          <BaseScript>
            <BaseValues>
              <DefaultIndex value="0"/>
              <!-- BaseCoordCount=2 -->
              <BaseCoord index="0" Format="1">
                <Coordinate value="-120"/>
              </BaseCoord>
              <BaseCoord index="1" Format="1">
                <Coordinate value="0"/>
              </BaseCoord>
            </BaseValues>
            <!-- BaseLangSysCount=0 -->
          </BaseScript>
        </BaseScriptRecord>
        <BaseScriptRecord index="5">
          <BaseScriptTag value="latn"/>
          <BaseScript>
            <BaseValues>
              <DefaultIndex value="1"/>
              <!-- BaseCoordCount=2 -->
              <BaseCoord index="0" Format="1">
                <Coordinate value="-120"/>
              </BaseCoord>
              <BaseCoord index="1" Format="1">
                <Coordinate value="0"/>
              </BaseCoord>
            </BaseValues>
            <!-- BaseLangSysCount=0 -->
          </BaseScript>
        </BaseScriptRecord>
      </BaseScriptList>
    </HorizAxis>
  </BASE>
```

File: Lib/fontTools/feaLib/testdata/spec9a.ttx (hunk 1 above)
Comment (khaledhosny) [inline Lib/fontTools/feaLib/testdata/spec9a.ttx]: I omitted the count since it will be calculated anyway when compiling the table.

**Comment by khaledhosny** [comment]
Four lines of code actually to get all the counts (including setting `BaseLangSysCount` to 0), I have them locally and can push them if you wish, but it feels redundant for not practical gain.

**Comment by anthrotype** [comment]
:+1: for printing the counts in the comment. We do the same elsewhere in TTX.

**Comment by khaledhosny** [comment]
OK, updated.

**Comment by brawer** [comment]
Thank you!

**Pull request merged**